src/components/PDFView/index.js

```diff
@@ -49,42 +104,60 @@ class PDFView extends PureComponent {
-                <Document
-                    loading={<FullScreenLoadingIndicator />}
-                    file={this.props.sourceURL}
-                    options={{
-                        cMapUrl: 'cmaps/',
-                        cMapPacked: true,
+        const outerContainerStyle = [
+            styles.w100,
+            styles.h100,
+            styles.justifyContentCenter,
+            styles.alignItemsCenter,
+        ];
+
+        // If we're requesting a password then we need to hide - but still render -
+        // the PDF component.
+        const pdfContainerStyle = this.state.shouldRequestPassword
+            ? [styles.PDFView, this.props.style, styles.invisible]
+            : [styles.PDFView, this.props.style];
 
-                    }}
-                    externalLinkTarget="_blank"
-                    onLoadSuccess={this.onDocumentLoadSuccess}
+        return (
+            <View style={outerContainerStyle}>
+                <View
+                    style={pdfContainerStyle}
+                    onLayout={event => this.setState({windowWidth: event.nativeEvent.layout.width})}
                 >
-                    {
-                        Array.from(
-                            new Array(this.state.numPages),
-                            (el, index) => (
-                                <Page
-                                    width={pageWidth}
-                                    key={`page_${index + 1}`}
-                                    pageNumber={index + 1}
-                                />
-                            ),
-                        )
-                    }
-                </Document>
+                    <Document
+                        loading={<FullScreenLoadingIndicator />}
+                        file={this.props.sourceURL}
+                        options={{
+                            cMapUrl: 'cmaps/',
+                            cMapPacked: true,
+                        }}
+                        externalLinkTarget="_blank"
+                        onLoadSuccess={this.onDocumentLoadSuccess}
+                        onPassword={this.initiatePasswordChallenge}
+                    >
+                        {_.map(_.range(this.state.numPages), (v, index) => (
+                            <Page
+                                width={pageWidth}
+                                key={`page_${index + 1}`}
+                                pageNumber={index + 1}
+                            />
+                        ))}
+                    </Document>
+                </View>
+                {this.state.shouldRequestPassword && (
+                    <PDFPasswordForm
+                        onSubmit={this.attemptPdfLoad}
+                        onPasswordUpdated={() => this.setState({isPasswordInvalid: false})}
+                        isPasswordInvalid={this.state.isPasswordInvalid}
+                        shouldAutofocusPasswordField={!this.props.isSmallScreenWidth}
```

Comment:
@tgolen @frenkield @parasharrajat @Santhosh-Sellavel 
We're coming from [this](https://github.com/Expensify/App/issues/14265) issue. Basically a request to autofocus password field on native devices. Any idea why this condition was added. Just want to clarify before we proceed with the linked issue. cc @sobitneupane

Reply:
I don't specifically remember either.